prototype2/efu/main.cpp

```diff
@@ -2,22 +2,26 @@
 
 #include <common/EFUArgs.h>
 #include <common/StatPublisher.h>
-#include <common/Trace.h>
+#include <common/Log.h>
 #include <common/Version.h>
 #include <efu/ExitHandler.h>
 #include <efu/HwCheck.h>
 #include <efu/Launcher.h>
 #include <efu/Loader.h>
 #include <efu/Parser.h>
 #include <efu/Server.h>
-#include <iostream>
 #include <libs/include/Timer.h>
 #include <libs/include/gccintel.h>
 #include <unistd.h> // sleep()
 #include <vector>
 
 #define ONE_SECOND_US 1000000U
 
+std::string ConsoleFormatter(const LogMessage &msg) {
```

Comment:
By the way, this formatting function should really have checks to make sure that the additional fields exist.